torch/testing/_asserts.py

```diff
@@ -98,6 +93,30 @@ def wrapper(
     return wrapper
 
 
+def _check_quantization(
+    check_tensor_values: Callable[..., Optional[_TestingErrorMeta]]
+) -> Callable[..., Optional[_TestingErrorMeta]]:
+    """Decorates continuous tensor check functions to handle quantized tensors.
+
+    If the inputs are not quantized, this decorator is a no-op.
+
+    Args:
+        check_tensor_values (Callable[..., Optional[Exception]]): Tensor check function for continuous tensors.
+
+    Returns:
+        Optional[Exception]: Return value of :attr:`check_tensors`.
+    """
+
+    @functools.wraps(check_tensor_values)
+    def wrapper(actual: Tensor, expected: Tensor, **kwargs: Any) -> Optional[_TestingErrorMeta]:
+        if not actual.is_quantized:
+            return check_tensor_values(actual, expected, **kwargs)
+
+        return check_tensor_values(actual.dequantize(), expected.dequantize(), **kwargs)
```

Comment:
As per @jerryzh168's request in , this now only checks for the `.dequantize()`'d version of the quantized tensors. Since this is agnostic with respect to the quantization scheme, we don't need any special handling here.